coreneuron/io/nrn_setup.cpp

```diff
@@ -1023,35 +1044,169 @@ size_t model_size(void) {
         printf("n_pntproc=%d sz=%ld nbyte=%ld\n", nt.n_pntproc, sz_pp, nt.n_pntproc * sz_pp);
         printf("n_netcon=%d sz=%ld nbyte=%ld\n", nt.n_netcon, sz_nc, nt.n_netcon * sz_nc);
         printf("n_weight = %d\n", nt.n_weight);
-#endif
 
-        // spike handling
-        nb_nt += nt.n_pntproc * sz_pp + nt.n_netcon * sz_nc + nt.n_presyn * sz_ps +
-                 nt.n_input_presyn * sz_psi + nt.n_weight * szd;
-        nbyte += nb_nt;
-#ifdef DEBUG
         printf("%d thread %d total bytes %ld\n", nrnmpi_myid, i, nb_nt);
 #endif
+
+        if (detailed_report) {
+            size_data[0] += nt.ncell;
+            size_data[1] += nt.end;
+            size_data[2] += nmech;
+            size_data[3] += nt._ndata;
+            size_data[4] += nt._nidata;
+            size_data[5] += nt._nvdata;
+            size_data[6] += nt.n_presyn;
+            size_data[7] += nt.n_input_presyn;
+            size_data[8] += nt.n_pntproc;
+            size_data[9] += nt.n_netcon;
+            size_data[10] += nt.n_weight;
+            size_data[11] += nb_nt;
+        }
     }
 
-#ifdef DEBUG
-    printf("%d netcon pointers %ld  nbyte=%ld\n", nrnmpi_myid, nccnt, nccnt * sizeof(NetCon*));
-#endif
     nbyte += nccnt * sizeof(NetCon*);
     nbyte += output_presyn_size();
     nbyte += input_presyn_size();
 
+    nbyte += nrnran123_instance_count() * nrnran123_state_size();
+
 #ifdef DEBUG
+    printf("%d netcon pointers %ld  nbyte=%ld\n", nrnmpi_myid, nccnt, nccnt * sizeof(NetCon*));
     printf("nrnran123 size=%ld cnt=%ld nbyte=%ld\n",
            nrnran123_state_size(),
            nrnran123_instance_count(),
            nrnran123_instance_count() * nrnran123_state_size());
+    printf("%d total bytes %ld\n", nrnmpi_myid, nbyte);
 #endif
+    if (detailed_report) {
+        size_data[12] = nbyte;
+#if NRNMPI
+        MPI_Allreduce(&size_data[0],
+                      &global_size_data_min[0],
+                      13,
+                      MPI_UNSIGNED_LONG_LONG,
+                      MPI_MIN,
+                      MPI_COMM_WORLD);
+        MPI_Allreduce(&size_data[0],
+                      &global_size_data_max[0],
+                      13,
+                      MPI_UNSIGNED_LONG_LONG,
+                      MPI_MAX,
+                      MPI_COMM_WORLD);
+        MPI_Allreduce(&size_data[0],
+                      &global_size_data_sum[0],
+                      13,
+                      MPI_UNSIGNED_LONG_LONG,
+                      MPI_SUM,
+                      MPI_COMM_WORLD);
```

Comment:
one more change that I couldn't think last time: could you use existing mpi wrappers from `coreneuron/mpi/mpispike.cpp` like `nrnmpi_long_allreduce_vec`? specially because they use duplicated comm `nrnmpi_comm` instead of hardcoding `MPI_COMM_WORLD`. (in the future they could be different when run coreneuron under NEURON)